controllers/manila_controller.go

```diff
@@ -508,27 +509,23 @@ func (r *ManilaReconciler) reconcileNormal(ctx context.Context, instance *manila
 	//
 	// check for required OpenStack secret holding passwords for service/admin user and add hash to the vars map
 	//
-	ospSecret, hash, err := secret.GetSecret(ctx, helper, instance.Spec.Secret, instance.Namespace)
+
+	result, err := verifyServiceSecret(
+		ctx,
+		types.NamespacedName{Namespace: instance.Namespace, Name: instance.Spec.Secret},
+		[]string{
+			instance.Spec.PasswordSelectors.Service,
+		},
+		helper.GetClient(),
+		&instance.Status.Conditions,
+		manila.NormalDuration,
+		&configVars,
+	)
 	if err != nil {
-		if k8s_errors.IsNotFound(err) {
-			r.Log.Info(fmt.Sprintf("OpenStack secret %s not found", instance.Spec.Secret))
-			instance.Status.Conditions.Set(condition.FalseCondition(
-				condition.InputReadyCondition,
-				condition.RequestedReason,
-				condition.SeverityInfo,
-				condition.InputReadyWaitingMessage))
-			return manila.ResultRequeue, nil
-		}
-		instance.Status.Conditions.Set(condition.FalseCondition(
-			condition.InputReadyCondition,
-			condition.ErrorReason,
-			condition.SeverityWarning,
-			condition.InputReadyErrorMessage,
-			err.Error()))
-		return ctrl.Result{}, err
+		return result, err
+	} else if (result != ctrl.Result{}) {
```

Comment:
@fmount so I very good with the current patchset , but not sure if you want to use the compound conditional that @abays says is fine. I don't think it's appropriate to mix the two approaches (compound conditional vs if..else), but if people agree the compound variant is cleaner then perhaps a global cleanup would be better in a follow-up patch.

tl;dr I'd be OK merging this as is (once CI completes)

Reply:
I agree it requires a cleanup wrt the mixed approach.
I'd be ok to follow up and make more code cleanup and dedup, so if that works and you're happy with the current PS, we can land this patch (if CI allows), and use the new common part as a reference for more code that we can deduplicate.

Reply:
Is the proposal then to tweak other repos to use the same pattern of a wrapper `verify*Secret()` function around `VerifySecret()` so that in the reconcile loops we can use a compound conditional?  Or are we just talking about making all the conditional in _this_ repo consistent?

Reply:
I have a couple of orthogonal responses.

The idea of using the compound conditional would apply to any existing code, not just the stuff added for verifying secrets. The idea just happened to come up in this PR.

The storage folks strive to make sure our operators (certainly glance, cinder, manila) are structurally similar. When something affects multiple operators we try to update all of them in as similar manner as possible. I'm working on a PR for the cinder-operator that will be nearly identical to manila's. Perhaps, when we get some time, one of us will submit a PR that replaces multiple if... else... code with the compound conditional, and if that gets accepted then we'll try to repeat it in the other storage operators.

That said, I would not expect EVERY other openstack operator adopt a similar pattern. Consistency across all operators is a great idea, but in practice it's too difficult to get everyone to agree on each pattern. We (storage) try to keep our own houses in order, and it's always nice to see other operators adopt an idea. But that's the extent of our evangelism.

Reply:
My hot take for cinder is https://github.com/openstack-k8s-operators/cinder-operator/pull/437. It's conceptually similar to this PR, although I organized the commits a little differently.